dwave_networkx/algorithms/markov.py

```diff
@@ -0,0 +1,205 @@
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+
+@binary_quadratic_model_sampler(1)
+def sample_markov_network(MN, sampler=None, fixed=None, return_sampleset=False,
+                          **sampler_args):
+    """Samples from a markov network using the provided sampler.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        A Markov Network as returned by :func:`.markov_network`
+
+    sampler
+        A binary quadratic model sampler. A sampler is a process that
+        samples from low energy states in models defined by an Ising
+        equation or a Quadratic Unconstrained Binary Optimization
+        Problem (QUBO). A sampler is expected to have a 'sample_qubo'
+        and 'sample_ising' method. A sampler is expected to return an
+        iterable of samples, in order of increasing energy. If no
+        sampler is provided, one must be provided using the
+        `set_default_sampler` function.
+
+    fixed : dict
+        A dictionary of variable assignments to be fixed in the markov network.
+
+    return_sampleset : bool (optional, default=False)
+        If True, returns a :obj:`dimod.SampleSet` rather than a list of samples.
+
+    **sampler_args
+        Additional keyword parameters are passed to the sampler.
+
+    Returns
+    -------
+    samples : list[dict]/:obj:`dimod.SampleSet`
+        A list of samples ordered from low-to-high energy or a sample set.
+
+    Examples
+    --------
+
+    >>> import dimod
+    ...
+    >>> potentials = {('a', 'b'): {
+        (0, 0): -1,
+        (0, 1): .5,
+        (1, 0): .5,
+        (1, 1): 2
+        }}
+    >>> MN = dnx.markov_network(potentials)
+    >>> sampler = dimod.ExactSolver()
+    >>> samples = dnx.sample_markov_network(MN, sampler)
+    >>> samples[0]
+    {'a': 0, 'b': 0}
+
+    >>> import dimod
+    ...
+    >>> potentials = {('a', 'b'): {
+        (0, 0): -1,
+        (0, 1): .5,
+        (1, 0): .5,
+        (1, 1): 2
+        }}
+    >>> MN = dnx.markov_network(potentials)
+    >>> sampler = dimod.ExactSolver()
+    >>> samples = dnx.sample_markov_network(MN, sampler, return_sampleset=True)
+    >>> samples.first
+    Sample(sample={'a': 0, 'b': 0}, energy=-1.0, num_occurrences=1)
+
+    Notes
+    -----
+    Samplers by their nature may not return the optimal solution. This
+    function does not attempt to confirm the quality of the returned
+    sample.
+
+    """
+
+    # we could do the fixing with dimod's FixVariablesComposite but for
+    # for backwards compatibility we'll do it on construction
+    bqm = markov_network_bqm(MN, fixed=fixed)
```

Comment:
@arcondello, Could you add an example (or modify a current one) to include fixed variables?

Reply:
done